### PR TITLE
Move models to `Microsoft.Docs.Template` razor classlib

### PR DESCRIPTION
**Pull request by yufeih**
This PR adds an `Microsoft.Docs.Template` project to host docs razor templates, models are moved to that project and made public.

#3050 #2579

### CODE REVIEW

File: src/Microsoft.Docs.Template/Models/TableOfContents.cs [hunk 1]
```diff
@@ -7,17 +7,27 @@
...
+    public class TableOfContentsModel
+    {
+        [JsonProperty(PropertyName = "metadata")]
+        public JObject Metadata { get; set; }
+
+        [JsonProperty(PropertyName = "items")]
+        public List<TableOfContentsItem> Items { get; set; }
+    }
+
+    public class TableOfContentsItem
```
Comment (live1206) [inline src/Microsoft.Docs.Template/Models/TableOfContents.cs]: >public class [](start = 4, length = 12)

Does StyleCop allow two public classes in one file?

File: src/Microsoft.Docs.Template/Models/GitUserInfo.cs [hunk 2]
```diff
@@ -3,7 +3,7 @@
 
 namespace Microsoft.Docs.Build
 {
-    internal class GitUserInfo
+    public class GitUserInfo
```
Comment (OsmondJiang) [inline src/Microsoft.Docs.Template/Models/GitUserInfo.cs]: >public [](start = 4, length = 6)

does this belongs to template model?

Comment (yufeih) [reply src/Microsoft.Docs.Template/Models/GitUserInfo.cs]: yes, PageModel uses this for author and contributors

---
In reply to: [210468146](https://github.com/dotnet/docfx/pull/3205#discussion_r210468146) [](ancestors = 210468146)

Comment (OsmondJiang) [reply src/Microsoft.Docs.Template/Models/GitUserInfo.cs]: why not put it inside PageModel?

---
In reply to: [210468265](https://github.com/dotnet/docfx/pull/3205#discussion_r210468265) [](ancestors = 210468265,210468146)

Comment (OsmondJiang) [reply src/Microsoft.Docs.Template/Models/GitUserInfo.cs]: I thought that's a output `page model`....

---
In reply to: [210468351](https://github.com/dotnet/docfx/pull/3205#discussion_r210468351) [](ancestors = 210468351,210468265,210468146)

File: src/docfx/models/TableOfContentsModel.cs [hunk 3]
```diff
@@ -1,18 +0,0 @@
...
-// Licensed under the MIT license. See LICENSE file in the project root for full license information.
-
-using System.Collections.Generic;
-using Newtonsoft.Json;
-using Newtonsoft.Json.Linq;
-
-namespace Microsoft.Docs.Build
-{
-    internal class TableOfContentsModel
-    {
```
Comment (OsmondJiang) [inline src/docfx/models/TableOfContentsModel.cs]: also delete the item class file?

File: src/Microsoft.Docs.Template/Models/LandingData.cs [hunk 4]
```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Docs.Build
+{
+    public class LandingData
```
[the remark this answers is not in the document]
Comment (yufeih) [reply src/Microsoft.Docs.Template/Models/LandingData.cs]: we could, I was just trying to blur the line here, as they are all just data models output by docfx and consumed by templates

---
In reply to: [210467431](https://github.com/dotnet/docfx/pull/3205#discussion_r210467431) [](ancestors = 210467431)

File: src/Microsoft.Docs.Template/Models/TableOfContents.cs (hunk 1 above)
Comment (live1206) [reply src/Microsoft.Docs.Template/Models/TableOfContents.cs]: Got it

---
In reply to: [210467560](https://github.com/dotnet/docfx/pull/3205#discussion_r210467560) [](ancestors = 210467560,210467136)

Comment (live1206) [reply src/Microsoft.Docs.Template/Models/TableOfContents.cs]: In V2, we have one file for JSON schema definitions, but for each JSON schema itself we have separate files, right?

---
In reply to: [210468771](https://github.com/dotnet/docfx/pull/3205#discussion_r210468771) [](ancestors = 210468771,210467560,210467136)

Comment (yufeih) [reply src/Microsoft.Docs.Template/Models/TableOfContents.cs]: yep

---
In reply to: [210468986](https://github.com/dotnet/docfx/pull/3205#discussion_r210468986) [](ancestors = 210468986,210468771,210467560,210467136)

**Pull request merged**